src/hydra-queue-runner/hydra-queue-runner.cc

```diff
@@ -449,74 +463,131 @@ bool State::checkCachedFailure(Step::ptr step, Connection & conn)
     return false;
 }
 
-
 void State::notificationSender()
 {
-    while (true) {
-        try {
+  using namespace std::string_literals;
+  std::optional<hydra_notify_state> hydra_notify = std::nullopt;
+
+  while (true) {
+    try {
+      bool should_fork = false;
```

Comment:
Two ways to refactor this:
* Refactor the code that checks whether we should fork into its own function, that returns true or false
* Refactor the code that actually does the forking into its own function, and call it instead of setting should_fork to true